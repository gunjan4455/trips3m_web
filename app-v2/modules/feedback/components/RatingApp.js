import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { RatingIcon,CloseIcon } from 'helpers/Icon/Icon';
import { ORGANISM_CLICKED, ORGANISM_VIEWED } from 'actions/segmentEvents';
import './Thankyou.scss';

class RatingApp extends Component {

  static propTypes = {
    trackSegmentEvent: PropTypes.func
  };

  componentDidMount() {
    this.trackSegment(ORGANISM_VIEWED, 'App Review Popup', '');
  }

  trackSegment = (event, section, cta) => {
    this.props.trackSegmentEvent({
      event,
      section,
      cta
    });
  };

  render() {
    return (
      <div className="flex flexDColumn text-center pt24 pb32 sbcw rating-popup relative">
        <a href="/" className="rating-cross" onClick={() => this.trackSegment(ORGANISM_CLICKED, 'App Review Popup', 'Close')}><CloseIcon /></a>
        <div className="relative star-rating">
          <RatingIcon />
        </div>
        <h3 className="fw9 f24 pt8 pb4 m0">Please rate our App and get INR 100 cash voucher!</h3>
        <p className="m0 f14p pb32 pl15 pr15">Please share the screenshot of the feedback on appfeedback@traveltriangle.com</p>

        <a className="btn-filled-pri rating-button"
           onClick={() => this.trackSegment(ORGANISM_CLICKED, 'App Review Popup', 'Rate Now')}
           href="https://c.singular.net/api/v1/ad?st=900297487404&h=b0c882ade525ff53572f2586d1d9c868fc3a3eaa">Rate
          Now</a>
      </div>
    );
  }
}

export default RatingApp;
