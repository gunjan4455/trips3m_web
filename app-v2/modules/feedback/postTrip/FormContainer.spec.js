import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import PostTripFormContainer from './FormContainer';
import RatingApp from "../components/RatingApp";

configure({ adapter: new Adapter() });

const mockcreatePostTripFeedback = jest.fn();
const mockPushState = jest.fn();

const DEFAULT_PROPS = {
  submitted: false,
  formValues: {},
  location: {
    query: {}
  },
  params: {
    requestedTripId: 4876448
  },
  createPostTripFeedback: mockcreatePostTripFeedback,
  pushState: mockPushState
};

const shallowWrapperPostTrip = (props) => {
  return shallow(<PostTripFormContainer {...props} />);
};

describe('Post trip module', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperPostTrip(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleSubmit properly', () => {
    const wrapper = shallowWrapperPostTrip(props);
    wrapper.instance().handleSubmit({});
    expect(mockcreatePostTripFeedback).toHaveBeenCalled();
  });

  it('should call setTimeout in componentwillreceiveprops function', () => {
    const wrapper = shallowWrapperPostTrip(props);
    jest.useFakeTimers();
    wrapper.setProps({ ...DEFAULT_PROPS, submitted: true });
    expect(wrapper.state().submitted).toBe(true);
    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 3000);
  });

  it('should show RatingApp popup when NPS rating is more than 7', () => {
    const wrapper = shallowWrapperPostTrip(props);
    wrapper.instance().setState({ npsScore: 8 });
    wrapper.setProps({ ...DEFAULT_PROPS, submitted: true });
    expect(wrapper.find(RatingApp)).toHaveLength(1);
  });
});
