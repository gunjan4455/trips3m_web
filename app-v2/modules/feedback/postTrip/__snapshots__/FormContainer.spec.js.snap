// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post trip module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostTripFormContainer
    createPostTripFeedback={[MockFunction]}
    formValues={Object {}}
    location={
      Object {
        "query": Object {},
      }
    }
    params={
      Object {
        "requestedTripId": 4876448,
      }
    }
    pushState={[MockFunction]}
    submitted={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ReduxForm
          enableReinitialize={true}
          formValues={Object {}}
          initialValues={
            Object {
              "nps": NaN,
            }
          }
          onSubmit={[Function]}
          trackSegmentEvent={[Function]}
        />,
        <ModalComponent
          customClass=""
          customCloseTrigger={false}
          disabled={false}
          fullView={false}
          hasCustomClose={false}
          hasNoClose={true}
          isOpen={false}
          isTriggerFullView={false}
          isWhiteIcon={false}
          onAfterModalOpen={[Function]}
          onRequestCloseCustom={[Function]}
          trigger={null}
        >
          <ThankYou />
        </ModalComponent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "enableReinitialize": true,
          "formValues": Object {},
          "initialValues": Object {
            "nps": NaN,
          },
          "onSubmit": [Function],
          "trackSegmentEvent": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ThankYou />,
          "customClass": "",
          "customCloseTrigger": false,
          "disabled": false,
          "fullView": false,
          "hasCustomClose": false,
          "hasNoClose": true,
          "isOpen": false,
          "isTriggerFullView": false,
          "isWhiteIcon": false,
          "onAfterModalOpen": [Function],
          "onRequestCloseCustom": [Function],
          "trigger": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ReduxForm
            enableReinitialize={true}
            formValues={Object {}}
            initialValues={
              Object {
                "nps": NaN,
              }
            }
            onSubmit={[Function]}
            trackSegmentEvent={[Function]}
          />,
          <ModalComponent
            customClass=""
            customCloseTrigger={false}
            disabled={false}
            fullView={false}
            hasCustomClose={false}
            hasNoClose={true}
            isOpen={false}
            isTriggerFullView={false}
            isWhiteIcon={false}
            onAfterModalOpen={[Function]}
            onRequestCloseCustom={[Function]}
            trigger={null}
          >
            <ThankYou />
          </ModalComponent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "enableReinitialize": true,
            "formValues": Object {},
            "initialValues": Object {
              "nps": NaN,
            },
            "onSubmit": [Function],
            "trackSegmentEvent": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ThankYou />,
            "customClass": "",
            "customCloseTrigger": false,
            "disabled": false,
            "fullView": false,
            "hasCustomClose": false,
            "hasNoClose": true,
            "isOpen": false,
            "isTriggerFullView": false,
            "isWhiteIcon": false,
            "onAfterModalOpen": [Function],
            "onRequestCloseCustom": [Function],
            "trigger": null,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
